src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.controllers;
+
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azure.storage.core.PathUtility;
+import io.restassured.RestAssured;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.File;
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(SpringRunner.class)
+@TestPropertySource("classpath:application.properties")
+public class EnvelopeDeletionTest {
+
+    @Value("${test-url}")
+    private transient String testUrl;
+
+    @Value("${scan-delay}")
+    private transient int scanDelay;
+
+    @Value("${storage-account-name")
+    private transient String accountName;
+
+    private transient CloudBlobContainer testContainer;
+
+    @Before
+    public void setUp() throws Exception {
+        String containerName = "test";
+        String sasToken = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .when()
+            .get(testUrl + "/token/" + containerName)
+            .thenReturn()
+            .jsonPath()
+            .getString("sas_token");
```

Comment:
Will retrieving SAS token work without confusing test service in config ?

Reply:
You should know better :grin:

Reply:
Added `test` service to configuration.